R/custom_disconnect_message.R

```diff
@@ -81,34 +93,27 @@ custom_disconnect_message <- function(refresh = "Refresh page",
       style = "display: none !important;",
       tags$div(
         id = "ss-connect-refresh",
-        tags$p("You've lost connection to the dashboard server - please try
-               refreshing the page:"),
-        tags$p(tags$a(
-          id = "ss-reload-link",
-          href = "#", "Refresh page",
-          onclick = "window.location.reload(true);"
-        )),
+        role = "alert",
+        tags$p(
+          "Sorry, you have lost connection to the", dashboard_title, "dashboard at the moment, please ",
+          tags$a(
+            id = "ss-reload-link",
+            href = "#", "refresh the page",
+            onclick = "window.location.reload(true);",
+            .noWS = c("after")
+          ),
+          "."
+        ),
         if (length(links) > 1) {
           tags$p(
-            "If this persists, you can also view the dashboard at one of our
-            mirror sites:",
-            tags$p(
-              tags$a(href = links[1], "Site 1"),
-              " - ",
-              tags$a(href = links[2], "Site 2"),
-              if (length(links) == 3) {
-                "-"
-              },
-              if (length(links) == 3) {
-                tags$a(href = links[3], "Site 3")
-              }
-            )
```

Comment:
This doesn't seem to handle more than one overflow site now, is that intentional?

If so we should make it clear in the params / description / validation. If not, it might be worth a couple of tests around this and then adding handling for more links back in?

Example here of a second mirror site not appearing in the HTML output (used KS4 as it would pass the URL validation)
```
> custom_disconnect_message(
+     dashboard_title = "DfE Shiny Template",
+     refresh = "Refresh page",
+     links = c(
+         "https://department-for-education.shinyapps.io/dfe-shiny-template",
+         "https://department-for-education.shinyapps.io/dfe-shiny-template-overflow/",
+         "https://department-for-education.shinyapps.io/KS4-transition-matrices/"
+     )
+ )
<script>$(function() {  $(document).on('shiny:disconnected', function(event) {    $('#custom-disconnect-dialog').show();    $('#ss-overlay').show();  })});</script>
<div id="custom-disconnect-dialog" style="display: none !important;">
  <div id="ss-connect-refresh" role="alert">
    <p>
      Sorry, you have lost connection to the
      DfE Shiny Template
      dashboard at the moment, please 
      <a id="ss-reload-link" href="#" onclick="window.location.reload(true);">refresh the page</a>.
    </p>
    <p>
      If you are still experiencing issues, please try our
      <a href="https://department-for-education.shinyapps.io/dfe-shiny-template-overflow/">alternative site</a>. Apologies for the inconvenience.
    </p>
    <p>
      Feel free to contact
      <a href="mailto:explore.statistics@education.gov.uk">explore.statistics@education.gov.uk</a>
      if you require further support.
    </p>
  </div>
</div>
<div id="ss-overlay" style="display: none;"></div>
```

Reply:
Yes, was intentional, should have added it in the PR description.

I think I set it up for many mirrors when we were first hitting the disconnect issues and we weren't sure how much mirrors would help / get used. From the analytics, I'm pretty sure they only do a relatively small amount of siphoning. Think it's still worth having the option for one, but don't see any evidence for that first mirror getting the levels of traffic that would make us ever want to have more than one mirror.

I did originally switch to the param just being overflow_link and having just a single URL, but stepped back from that given it was another breaking change and we've already built a few of those up recently.

I've added a validation warning in and improved the docs.

Reply:
Agree on no evidence more is useful - happy with this